### PR TITLE
all metadata is used by read_json

### PR DESCRIPTION
**Pull request by abhinavralhan**
- [ ] Tests added / passed
- [ ] Passes `flake8 dask`

### CODE REVIEW

**Comment by abhinavralhan** [comment]
@TomAugspurger thanks for the review. made the changes.

**Comment by TomAugspurger** [comment]
Thanks. Can you add tests for user-provided meta? Maybe look at the stack overflow question for inspiration.

Also, a lint failure: https://travis-ci.org/dask/dask/jobs/503147949#L1954

**Comment by abhinavralhan** [comment]
@TomAugspurger Corrected the lint failure and added a test case.

File: dask/dataframe/io/tests/test_json.py [hunk 1]
```diff
@@ -34,6 +34,10 @@ def test_read_json_basic(orient):
             out.columns = list(df.columns)
         assert_eq(out, df)
 
+        with pytest.raises(ValueError) as e:
```
Comment (TomAugspurger) [inline dask/dataframe/io/tests/test_json.py]: Let's split this into its own test specifically for `meta`.

I think the rough goal of this test should be to

1. create a JSON string that `dd.read_json` interprets incorrectly by default. The easiest way to do this is probably to have a column where the first partition is all ints, and the second partition is a mix of ints and NaNs.
2. Let the user specify `meta=...`, with the correct metadata.
3. Ensure that that *doesn't'* raise.

LMK if you need help with writing this, though it could be a little while before I can take a closer look.

Also, some test failures in https://travis-ci.org/dask/dask/jobs/503501340. I haven't looked too closely yet.

Comment (abhinavralhan) [reply dask/dataframe/io/tests/test_json.py]: By split into its own test, do you mean a new module for meta or a new file `test_meta`.

Can you show me an example of how user specified meta would look like? I was having confusion with that.
Also, I don't completely understand the CI error.

**Comment by TomAugspurger** [comment]
I mean a new test within the same file.

I can't make an example right now, but if you look through the tests for
dd.read_csv you may find inspiration.

File: dask/dataframe/tests/test_dataframe.py [hunk 2]
```diff
@@ -21,6 +21,7 @@
 from dask.dataframe import methods
 from dask.dataframe.utils import (assert_eq, make_meta, assert_max_deps,
                                   PANDAS_VERSION)
+warnings.simplefilter(action='ignore', category=FutureWarning)
```
Comment (TomAugspurger) [inline dask/dataframe/tests/test_dataframe.py]: I don't think we want this. Rather, we should

1. assert that the warning is emitted if we're the ones emitting the warning
2. ensure that the code in `dask/dataframe` doesn't case a warning.

File: dask/dataframe/tests/test_ufunc.py [hunk 3]
```diff
@@ -11,6 +10,8 @@
 import dask.dataframe as dd
 from dask.dataframe.utils import assert_eq
 
+pd = pytest.importorskip('pandas')
```
Comment (TomAugspurger) [inline dask/dataframe/tests/test_ufunc.py]: Likewise, I don't think we want this here.
